doc/website/release-notes/iceoryx-unreleased.md

```diff
@@ -139,3 +139,37 @@
         iox::posix::waitForTerminationRequest();
     }
     ```
+
+6. It is not possible to delete a class which is derived from `NewType` via a pointer to `NewType`
+
+   ```cpp
+   struct Foo : public iox::cxx::NewType<uint64_t, iox::cxx::newtype::ConstructByValueCopy>
+   {
+       using ThisType::ThisType;
+   };
+
+   iox::cxx::NewType<uint64_t, iox::cxx::newtype::ConstructByValueCopy>* soSmart = new Foo{42};
+
+   delete soSmart; // <- not possible anymore
+   ```
+
+7. It is not possible to use the `NewType` to create type aliases. This was not recommended and is now enforced
+
+   ```cpp
+   // before
+   // for the compiler Foo and Bar are the same type
+   using Foo = iox::cxx::NewType<uint64_t, iox::cxx::newtype::ConstructByValueCopy>;
+   using Bar = iox::cxx::NewType<uint64_t, iox::cxx::newtype::ConstructByValueCopy>;
+
+   // after
+   // compile time error when Foo and Bar are mixed up
+   struct Foo : public iox::cxx::NewType<uint64_t, iox::cxx::newtype::ConstructByValueCopy>
+   {
+       using ThisType::ThisType;
+   };
+
+   struct Bar : public iox::cxx::NewType<uint64_t, iox::cxx::newtype::ConstructByValueCopy>
+   {
+       using ThisType::ThisType;
+   };
```

Comment:
What do you think about such a macro?
```cpp
#define IOX_NEW_TYPE(TypeName, Type, ...)                            \
    struct TypeName : public iox::cxx::NewType<Type, __VA_ARGS__>    \
    {                                                                \
        using ThisType::ThisType;                                    \
        using ThisType::operator=;                                   \
                                                                     \
        TypeName() noexcept = default;                               \
        TypeName(const TypeName&) noexcept = default;                \
        TypeName(TypeName&&) noexcept = default;                     \
        TypeName& operator=(const TypeName&) noexcept = default;     \
        TypeName& operator=(TypeName&&) noexcept = default;          \
    };

IOX_NEW_TYPE(Foo, uint64_t, iox::cxx::newtype::ConstructByValueCopy)
IOX_NEW_TYPE(Bar, uint64_t, iox::cxx::newtype::ConstructByValueCopy)
```

If you like it, I can create an issue an implement it.

Reply:
I think we do not have a use case in iceoryx where such a macro would be used at the moment. I think it is quite often the case that we inherit from newtype and add some additional methods, see `UniquePortId` or customize the constructor see `UniqueId`.

Reply:
@elfenpiff maybe it is not used because the threshold is higher than necessary ;)

Reply:
Then add this macro :D